osu.Game/Overlays/Toolbar/ToolbarButton.cs

```diff
@@ -222,6 +201,20 @@ public void OnReleased(GlobalAction action)
 
         private void updateKeyBindingTooltip()
         {
+            if (Hotkey == null) return;
+
+            var realmKeyBinding = realmFactory.Context.All<RealmKeyBinding>().FirstOrDefault(rkb => rkb.RulesetID == null && rkb.ActionInt == (int)Hotkey.Value);
+
+            // TODO: temporarily disabled to avoid crashes when querying after ExecutionState is changed.
+            // if (realmKeyBinding != null)
+            // {
+            //     realmKeyBinding.PropertyChanged += (sender, args) =>
+            //     {
+            //         if (args.PropertyName == nameof(realmKeyBinding.KeyCombinationString))
+            //             updateKeyBindingTooltip();
+            //     };
+            // }
```

Comment:
Not sure about moving this code into a comment here - it's a bit of a weird one at face value, especially that it looks like the end goal is to have this method call itself recursively on property change. I'd probably just remove and rely on git history to get that back once the issues are all fixed there.